src/gf.c

```diff
@@ -2500,9 +2511,8 @@ static int ml_matches_visitor(jl_typemap_entry_t *ml, struct typemap_intersectio
                         // the current method doesn't match if there is an intersection with an
                         // ambiguous method that covers our intersection with this one.
                         jl_value_t *ambi = jl_type_intersection_matching((jl_value_t*)ml->sig,
-                                                                         (jl_value_t*)mambig->sig,
-                                                                         &env, mambig->tvars);
-                        if (jl_subtype(closure->match.ti, ambi, 0)) {
+                                                                         (jl_value_t*)mambig->sig, &env);
+                        if (jl_subtype(closure->match.ti, ambi)) {
```

Comment:
I think the correct version should read more as a tautology: the current method won't match at runtime if another method would also match at runtime over the entire computed type intersection.

```c
if (jl_subtype(closure->match.ti, mambig->sig))
```